CitationStyleLanguagePlugin.php

```diff
@@ -0,0 +1,1048 @@
+                        str_replace('_', '-', substr(Locale::getLocale(), 0, 5)),
+                        substr(Locale::getLocale(), 0, 2),
+                        'en-US',
+                    ] as $tryLocale) {
+                        if (file_exists(dirname(__FILE__) . '/lib/vendor/citation-style-language/locales/locales-' . $tryLocale . '.xml')) {
+                            break;
+                        }
+                    }
+
+                    //Clickable URL and DOI including affixes
+                    $additionalMarkup = [
+                        'DOI' => [
+                            'function' => function ($cslItem, $renderedText) {
+                                return '<a href="https://doi.org/' . $cslItem->DOI . '">' . $renderedText . '</a>';
+                            },
+                            'affixes' => true
+                        ],
+                        'URL' => [
+                            'function' => function ($cslItem, $renderedText) {
+                                return '<a href="' . $cslItem->URL . '">' . $renderedText . '</a>';
+                            },
+                            'affixes' => true
+                        ],
+                    ];
+
+                    $citeProc = new CiteProc($style, $tryLocale, $additionalMarkup);
+                    $citation = $citeProc->render([$citationData], 'bibliography');
+                }
+            }
+        }
+
+        return $citation;
+    }
+
+    /**
+     * Load a CSL style and return the contents as a string
+     *
+     * @param $styleConfig array CSL configuration to load
+     */
+    public function loadStyle($styleConfig)
+    {
+        if (!empty($styleConfig['useCsl'])) {
+            return file_get_contents($styleConfig['useCsl']);
+        } else {
+            return file_get_contents($this->getPluginPath() . '/citation-styles/' . $styleConfig['id'] . '.csl');
+        }
+    }
+
+    /**
+     * Download a citation format
+     *
+     * Downloadable citation formats can be used to import into third-party
+     * software.
+     *
+     * @param $request Request
+     * @param $submission Submission
+     * @param $citationStyle string Name of the citation style to use.
+     * @param $issue Issue Optional. Will fetch from db if not passed.
+     * @param $publication Publication Optional.
+     * @param $chapter Chapter Optional. OMP chapter pages only.
+     *
+     * @return string
+     */
+    public function downloadCitation($request, $submission, $citationStyle = 'ris', $issue = null, $publication = null, $chapter = null)
+    {
+        if (($this->application === 'ojs2') && empty($issue)) {
+            $issue = Repo::issue()->get($submission->getCurrentPublication()->getData('issueId'));
+        }
+
+        $styleConfig = $this->getCitationStyleConfig($citationStyle);
+        if (empty($styleConfig)) {
+            return false;
+        }
+
+        if (!$publication) {
```

Comment:
I think this `if` is new; shouldn't the `$publication` always be provided? (This is another example of something we're working to tidy up -- if we know something is always provided, a `Publication $publication` typehint will prove it; if it's nullable, a `Publication ?$publication` will document it.)